packages/cliqz-home-react/src/SpeedDialsRow.js

```diff
@@ -0,0 +1,160 @@
+    dial.removedAt = index;
+    this.state.removedItems.push(dial);
+    this.setState({
+      removedItems: this.state.removedItems
+    });
+
+    cliqz.core.sendTelemetry({
+      type: 'home',
+      action: 'click',
+      target_type: 'remove-' + (dial.custom ? 'custom' : 'history'),
+      target_index: index
+    });
+  }
+
+  closeUndo(type) {
+    this.setState({
+      removedItems: []
+    });
+
+    cliqz.core.sendTelemetry({
+      type: 'home',
+      action: 'click',
+      target_type: 'close-' + type + '-undo'
+    });
+  }
+
+  _reAddSpeedDial(dial) {
+    const index = dial.removedAt;
+    this.props.addSpeedDial(dial, index);
+    cliqz.freshtab.addSpeedDial(dial.url, index).then(() => {
+      let newItems = this.state.removedItems.filter((item) => {
+        return item != dial;
+      });
+      this.setState({
+        removedItems: newItems
+      });
+    })
+  }
+
+  undoRemoval() {
+    const speedDial = this.state.removedItems.pop();
+    const type = speedDial.custom ? 'custom' : 'history';
+
+    if(speedDial.custom) {
+      this._reAddSpeedDial.call(this, speedDial);
+    } else {
+      cliqz.freshtab.revertHistorySpeedDial(speedDial.url);
+      this.props.addSpeedDial(speedDial, speedDial.removedAt);
+    }
+
+    cliqz.core.sendTelemetry({
+      type: 'home',
+      action: 'click',
+      target_type: 'undo-' + type + '-remove'
+    });
+  }
+
+  resetAll() {
+    cliqz.freshtab.resetAllHistory().then( results => {
+      this.closeUndo.call(this, 'history');
+      this.props.getSpeedDials();
+    })
+
+    cliqz.core.sendTelemetry({
+      type: 'home',
+      action: 'click',
+      target_type: 'reset-all-history'
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <h3 className="speedDialLabel">Label to be localized</h3>
+        <div className="speed-dial-row">
+          {
+            this.props.dials.slice(0,5).map((dial, i) => {
+              return <SpeedDial 
+                        key={i} 
+                        dial={dial}
+                        removeSpeedDial={this.removeSpeedDial.bind(this, dial, i)} />
+            })
+          }
+
+          {this.state.showAddButton() &&
+            <AddSpeedDial 
+              addSpeedDial={this.props.addSpeedDial} />
+          }
+        </div>
+        <div className="undoOptions">
+          {this.state.removedItems.length > 0 &&
+            <div>
+              <span>
+                {this.state.removedItems.length}
+                site was removed
+              </span>
+              <a className="undo"
+                 href="#"
+                 onClick={this.undoRemoval}>
+                 Undo
+              </a>
+              <a className="resetAll"
+                 href="#"
+                 onClick={this.resetAll}>
+                 Reset all
+              </a>
+              <a className="closeUndo"
+                 onClick={this.closeUndo.bind(this, this.props.type)}>
```

Comment:
wonder which patter in nicer, this or `onClick={() => this.closeUndo(this.props.type)}` or rather `onClick={() => this.onClick()}`

the last one is the abbervation of `onClick={this.onClick}`, assuming `onClick` is bound to this - which opens a question if we should use bind in JSX components